faculty_cli/cli.py

```diff
@@ -1451,3 +1450,147 @@ def dataset_ls(project, prefix, show_hidden):
         prefix, project_id=project_id, show_hidden=show_hidden
     ):
         click.echo(item)
+
+
+@cli.group()
+def template():
+    """Manipulate templates in the Faculty knowledge centre."""
+    pass
+
+
+@template.command()
+def init():
+    """Create a blank template."""
+    print("hello")
+
+
+@template.command()
+@click.argument(
+    "source_template_name"
+)  # Outstanding question as to whether this should also alllow the template_ID
+@click.option(
+    "--version",
+    default=None,
+    help="The specific version of the template to use.",
+)
+@click.option(
+    "--directory",
+    default=Path.cwd(),
+    help="The optional target directory where the template will be placed.",
+    type=click.Path(),
+)
+def clone(source_template_name, version, directory):
+    """Clone an existing template."""
+    print(source_template_name)
+    print(directory)
+
+    if version is None:
+        # TODO: Get the latest version
+        print(version)
+
+
+@template.command()
+@click.argument("project_identifier")
+@click.option(
+    "--source_directory",
+    default=Path.cwd(),
+    type=click.Path(),
+    help="Path to the source template directory where the template"
+    + "will be placed.",
+)
+@click.option(
+    "--target-directory",
+    default=Path("/"),
+    help="The specific version of the template to use.",
+)
+@click.option(
+    "parameters",
+    "--parameter",
+    "-p",
+    type=(str, str),
+    default=[],
+    help="The optional target directory where the template will be placed.",
+    multiple=True,
+)
+def apply_from_directory(
+    project_identifier, source_directory, target_directory, parameters
+):
+    """Apply from a directory to a project."""
+    project_id = _resolve_project(project_identifier)
+    print(project_id)
+    print(source_directory)
+    print(target_directory)
+
+    for (key, value) in parameters:
+        print(key, value)
+
+
+@template.command()
+@click.argument("project_name")
+@click.option(
+    "--source_directory",
+    default=Path.cwd(),
+    type=click.Path(),
+    help="Path to the source template directory where the template."
+    + "will be placed.",
+)
+@click.option(
+    "--target-directory",
+    default=Path("/"),
+    help="The specific version of the template to use.",
+)
+@click.option(
+    "parameters",
+    "--parameter",
+    "-p",
+    type=(str, str),
+    default=[],
+    help="The optional target directory where the template will be placed.",
+    multiple=True,
+)
+def create_project_from_directory(
+    project_name, source_directory, target_directory, parameters
+):
+    """Apply from a directory to a new project."""
+    print(project_name)
+    print(source_directory)
+    print(target_directory)
+
+    for (key, value) in parameters:
+        print(key, value)
+
+
+@template.group()
+def publish():
+    """Publish to the knowledge centre."""
+    pass
+
+
+@publish.command(name="new")
+@click.argument("template_name")
+@click.option(
```

Comment:
As above, this should be a positional argument.

Reply:
Good spot.